source/base/data_out_base.cc

```diff
@@ -2947,20 +2962,11 @@ namespace DataOutBase
           << "#" << '\n'
           << "# ";
 
-      switch (spacedim)
+      AssertThrow(spacedim <= flags.space_dimension_labels.size(),
+                  GnuplotFlags::ExcNotEnoughSpaceDimensionLabels());
+      for (unsigned int spacedim_n=0; spacedim_n<spacedim; ++spacedim_n)
         {
-        case 1:
-          out << "<x> ";
-          break;
-        case 2:
-          out << "<x> <y> ";
-          break;
-        case 3:
-          out << "<x> <y> <z> ";
-          break;
-
-        default:
-          Assert (false, ExcNotImplemented());
+          out << '<' << flags.space_dimension_labels.at(spacedim_n) << "> ";
```

Comment:
I do not feel strongly about this but I prefer the bounds checked ones since the extra safety doesn't really cost much extra.